include/highfive/bits/H5Slice_traits.hpp

```diff
@@ -23,7 +23,28 @@ class SliceTraits;
 
 class ElementSet {
   public:
+    ///
```

Comment:
I think we should template ElementSet to the number of dimensions and use `std::array`
`ElementSet(std::initializer_list<std::array<std::size_t, NDIM>> list);`